app/src/test/java/supermarket/ProductTest.kt

```diff
@@ -1,84 +1,315 @@
 package supermarket
 
-import org.hamcrest.CoreMatchers.`is` as Is
+import junit.framework.Assert.assertTrue
+import org.hamcrest.CoreMatchers.containsString
 import org.hamcrest.MatcherAssert.assertThat
-import org.hamcrest.Matchers.closeTo
+import org.junit.After
 import org.junit.Before
 import org.junit.Test
 import supermarket.TestUtils.apples
+import supermarket.TestUtils.flour
+import supermarket.TestUtils.oranges
 import supermarket.TestUtils.toothbrush
 import supermarket.TestUtils.toothpaste
 import supermarket.model.BundleOffer
 import supermarket.model.PercentageOffer
+import supermarket.model.QuantityForAmountOffer
+import supermarket.model.Receipt
 import supermarket.model.ShoppingCart
 import supermarket.model.Teller
 import supermarket.model.ThreeForTwoOffer
+import org.hamcrest.CoreMatchers.`is` as Is
 
 class ProductTest {
     private val catalog = FakeCatalog()
     private val cart = ShoppingCart()
     private val teller = Teller(catalog)
+    private lateinit var receipt: Receipt
 
     @Before
     fun setup() {
         TestUtils.setupCatalog(catalog)
     }
 
-    @Test
-    fun `toothbrush discount is recognized`() {
-        cart.addItemQuantity(toothbrush, 1.0)
+    @After
+    fun tearDown() {
+        // No teardown necessary, but print receipt while we're at it.
```

Comment:
Since there is no teardown needed or occurring, couldn't this name just be something like `printReciept()`with the after annotation since that's all it's doing.

Reply:
That's fair.  I mostly liked the symmetry of having a setup & teardown function.